pkg/detector/detector.go

```diff
@@ -119,7 +128,7 @@ func (d *ResourceDetector) Start(ctx context.Context) error {
 	d.InformerManager.ForResource(propagationPolicyGVR, d.EventHandler)
 	d.InformerManager.ForResource(clusterPropagationPolicyGVR, d.EventHandler)
 	d.Processor = util.NewAsyncWorker("resource detector", ClusterWideKeyFunc, d.Reconcile)
-	d.Processor.Run(1, d.stopCh)
+	d.Processor.Run(concurrentReconciles, d.stopCh)
```

Comment:
This is a regression as @RainbowMango mentioned [here](https://github.com/karmada-io/karmada/pull/1199#pullrequestreview-860571089)

Reply:
+1 
We can't start more than one worker before this risk is solved.